src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/matcher/SymbolicRegexMatcher.cs

```diff
@@ -496,10 +508,9 @@ private State<S> CreateNewTransition(State<S> q, S atom, int offset)
             }
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void CheckTimeout()
         {
-            if (Environment.TickCount > _timeoutOccursAt)
+            if (System.Environment.TickCount > _timeoutOccursAt)
```

Comment:
because TickCount is int, it will overflow at some random point within the next 49 days, so I think you need to do the same logic that is done here
https://github.com/dotnet/runtime/blob/e7b743cae325d4720fceaf7f84f68aaaf7bb7dbe/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs#L376-L400

Reply:
we can tune the efficiency later, but that's a correctness thing so if you don't fix it, we should add a todo.

Reply:
Ok, I'll copy the functionality from there and will do the same logic in srm. I'll add this as a fix to this pr.